overlord/restart/restart.go

```diff
@@ -534,8 +546,14 @@ func FinishTaskWithRestart(t *state.Task, status state.Status, restartType Resta
 
 // PendingForChange checks if a system restart is pending for a change.
 func PendingForChange(st *state.State, chg *state.Change) bool {
+	return PendingForChangeConsidering(st, chg, nil)
+}
+
+// PendingForChangeConsidering checks if a system restart is pending for a
+// change, ignoring tasks other than the task IDs supplied.
+func PendingForChangeConsidering(st *state.State, chg *state.Change, considerTasks map[string]bool) bool {
```

Comment:
wouldn't PendingForChangeTasks be a simpler name?  also shouldn't this have a unit test in the package

Reply:
I've added the missing test now and simplified the name